2009-11-07	Magnus Granberg		<zorry@ume.nu>

		bug #292139
		* Makeconfig	+link-pie:	set +link to +link-pie
		+link-static:	change $(static-start-installed-name) to S$(static-start-installed-name)
		change +prector to +prectorS, change +postctor to +postctorS
		+link-bounded: change +prector to +prectorS, change +postctor to +postctorS

--- Makeconfig	2009-11-06 16:39:18.000000000 +0100
+++ Makeconfig	2009-11-07 16:39:35.000000000 +0100
@@ -447,29 +447,30 @@
 	     $(common-objpfx)libc% $(+postinit),$^) \
 	     $(link-extra-libs) $(link-libc) $(+postctorS) $(+postinit)
 endif
++link = $(+link-pie)
 # Command for statically linking programs with the C library.
 ifndef +link-static
 +link-static = $(CC) -nostdlib -nostartfiles -static -o $@ \
 	      $(sysdep-LDFLAGS) $(LDFLAGS) $(LDFLAGS-$(@F))  \
-	      $(addprefix $(csu-objpfx),$(static-start-installed-name)) \
-	      $(+preinit) $(+prector) \
+	      $(addprefix $(csu-objpfx),S$(static-start-installed-name)) \
+	      $(+preinit) $(+prectorS) \
 	      $(filter-out $(addprefix $(csu-objpfx),start.o \
 						     $(start-installed-name))\
 			   $(+preinit) $(link-extra-libs-static) \
 			   $(common-objpfx)libc% $(+postinit),$^) \
-	      $(link-extra-libs-static) $(link-libc-static) $(+postctor) $(+postinit)
+	      $(link-extra-libs-static) $(link-libc-static) $(+postctorS) $(+postinit)
 endif
 # Command for statically linking bounded-pointer programs with the C library.
 ifndef +link-bounded
 +link-bounded = $(CC) -nostdlib -nostartfiles -static -fbounded-pointers -o $@ \
 	      $(sysdep-LDFLAGS) $(LDFLAGS)  \
 	      $(addprefix $(csu-objpfx),b$(static-start-installed-name)) \
-	      $(+preinit) $(+prector) \
+	      $(+preinit) $(+prectorS) \
 	      $(filter-out $(addprefix $(csu-objpfx),start.ob \
 						     $(start-installed-name))\
 			   $(+preinit) $(link-extra-libs-bounded) \
 			   $(common-objpfx)libc% $(+postinit),$^) \
-	      $(link-extra-libs-bounded) $(link-libc-bounded) $(+postctor) $(+postinit)
+	      $(link-extra-libs-bounded) $(link-libc-bounded) $(+postctorS) $(+postinit)
 endif
 ifndef config-LDFLAGS
 ifeq (yesyes,$(build-shared)$(elf))
